File: labs/lab4/task_2_2/src/NorthAmericanCity.java
public class NorthAmericanCity implements City {

    private final String name;
    private final double temperature;
    private boolean hasWeatherWarning;

    public NorthAmericanCity(String name, double temperature) {
        this.name = name;
        this.temperature = temperature;
        hasWeatherWarning = false;
    }

    public String getName() {
        return name;
    }

    public double getTemperature() {
        return temperature;
    }

    public String getTemperatureScale() {
        return "Fahrenheit";
    }

    public boolean getHasWeatherWarning() {
        return hasWeatherWarning;
    }

    public void setHasWeatherWarning(boolean hasWeatherWarning) {
        this.hasWeatherWarning = hasWeatherWarning;
    }

}

File: Python/labs/lab4/task_2_2/NorthAmericanCity.py
from City import City

class NorthAmericanCity(City):

    def __init__(self, name, temperature):
        self.name = name
        self.temperature = temperature
        self.hasWeatherWarning = False

    def getName(self):
        return self.name

    def getTemperature(self):
        return self.temperature

    def getTemperatureScale(self):
        return "Fahrenheit"

    def getHasWeatherWarning(self):
        return self.hasWeatherWarning

    def setHasWeatherWarning(self, hasWeatherWarning):
        self.hasWeatherWarning = hasWeatherWarning